caput/astro/coordinates/_spherical.pyi

```diff
@@ -0,0 +1,167 @@
+import numpy as np
+import numpy.typing as npt
+
+__all__ = [
+    "cart_to_sph",
+    "cosine_rule",
+    "great_circle_points",
+    "norm_vec2",
+    "sph_dot",
+    "sph_to_cart",
+    "thetaphi_plane",
+    "thetaphi_plane_cart",
+]
+
+#
+def sph_to_cart(
+    sph_coords: npt.NDArray[np.integer | np.floating],
+) -> np.ndarray[np.float64]:
+    """Convert a vector in Spherical Polar coordinates to Cartesians.
+
+    This routine is OpenMP parallel.
+
+    Parameters
+    ----------
+    sph_coords
+        A vector (or array of) in spherical polar co-ordinates. Values should be
+        packed as [r, theta, phi] along the last axis. Alternatively they can be
+        packed as [theta, phi] in which case r is assumed to be one.
+
+    Returns
+    -------
+    Array of equivalent vectors in cartesian coordinartes.
+    """
+    ...
+
+#
+def cart_to_sph(
+    cart_arr: npt.NDArray[np.integer | np.floating],
+) -> np.ndarray[np.floating]:
+    """Convert a cartesian vector into Spherical Polars.
+
+    Uses the same convention as `sph_to_cart`.
+
+    Parameters
+    ----------
+    cart_arr
+        Array of cartesians.
+
+    Returns
+    -------
+    Array of spherical polars (packed as [[ r1, theta1, phi1], [r2, theta2,
+    phi2], ...]
+    """
+    ...
+
+#
+def sph_dot(
+    arr1: npt.NDArray[np.integer | np.floating],
+    arr2: npt.NDArray[np.integer | np.floating],
+) -> np.ndarray[np.floating]:
+    """Take the scalar product in spherical polars.
+
+    Parameters
+    ----------
+    arr1, arr2
+        Two arrays of vectors in spherical polars [theta, phi], (or
+        alternatively as [theta, phi]). Should be broadcastable against each
+        other.
+
+    Returns
+    -------
+    An array of the dotted vectors.
+    """
+    ...
+
+#
+def thetaphi_plane(
+    sph_arr: npt.NDArray[np.integer | np.floating],
+) -> tuple[np.ndarray[np.floating], np.ndarray[np.floating]]:
+    """For each position, return the theta, phi unit vectors (in spherical polars).
+
+    Parameters
+    ----------
+    sph_arr
+        Angular positions (in spherical polar co-ordinates).
+
+    Returns
+    -------
+    Unit vectors in the theta and phi directions (still in spherical
+    polars).
```

Comment:
```suggestion
    Theta and phi directions (still in spherical polar co-ordinates) for the unit vectors.
```

The point here (perhaps poorly executed) is to emphasise that this function returns a 2-tuple of arrays of angles, not a single array of angle pairs.